aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -421,8 +424,19 @@ internal class RealAWSCognitoAuthPlugin(
                 }
             },
             {
+                val signInOptions = if (options !is AWSCognitoAuthSignInOptions) {
+                    AWSCognitoAuthSignInOptions.builder().authFlowType(AuthFlowType.USER_SRP_AUTH).build()
+                } else {
+                    options
+                }
+
                 val event = AuthenticationEvent(
-                    AuthenticationEvent.EventType.SignInRequested(username, password, options)
+                    AuthenticationEvent.EventType.SignInRequested(
+                        username,
+                        password,
+                        signInOptions.authFlowType.toString(),
+                        signInOptions.metadata
```

Comment:
`metadata` is not part of AuthSignInOptions. This will throw an error.

Reply:
This will not throw an error. If you see the line above: 

```
 val signInOptions = if (options !is AWSCognitoAuthSignInOptions) {
                    AWSCognitoAuthSignInOptions.builder().authFlowType(AuthFlowType.USER_SRP_AUTH).build()
                } else {
                    options
                }
```

Reply:
If options is not of type AWSCognitoAuthSignInOptions, then it is of type AuthSignInOptions which does not have metadata.

Reply:
Removed and moved the check to Actions.